Sources/NIOHTTP2/MultiplexerAbstractChannel.swift

```diff
@@ -89,24 +90,29 @@ extension MultiplexerAbstractChannel {
         }
     }
 
-    func configureInboundStream(initializer: InboundStreamStateInitializer) {
+    func configureInboundStream(initializer: InboundStreamStateInitializer, promise: EventLoopPromise<Any>?) {
         switch initializer {
         case .includesStreamID(let initializer):
             self.baseChannel.configure(initializer: initializer, userPromise: nil)
         case .excludesStreamID(let initializer):
             self.baseChannel.configure(initializer: initializer, userPromise: nil)
+        case .returnsAny(let initializer):
+            self.baseChannel.configure(initializer: initializer, userPromise: promise)
```

Comment:
If `promise` isn't `nil` then it will be leaked if the initialiser is one of the other two cases.

Reply:
I've separated out the configuration function with a precondition to ensure that the promise-supplying case only accepts `.returnsAny`. I _could_ wire it up with a type transformation on the other promises but in practice they are never used at the moment.